portfolio/src/main/webapp/script.js

```diff
@@ -28,6 +28,15 @@ function addRandomQuote() {
   quoteContainer.innerText = quote;
 }
 
+/*
+ * Adds the data from /data to page
+ */
+function addGreetingFromData() {
+    fetch('/data').then(response => response.text()).then((greeting) => {
+        document.getElementById('greeting-container').innerHTML = greeting;
```

Comment:
Not essential here, but some things to think about if you have time:

- do you want this to have some kind of prompt while /data is loading?
- what do you want to happen when /data errors out (e.g. site is temporarily down, gremlins got in the network)? Right now, it will console.log and you will get no output

Reply:
I think we are changing this function in the later sections, but in these cases I think we probably would want a loading message in the console or on screen while /data is loading, maybe by checking the status of the promise. Also if the site is down then we would want to print and error message on the page/log too. I will probably look into these after finishing the walkthrough steps

Reply:
Ah, sounds good!

I think those are reasonable options. You would probably want the loading message on screen, though; users don't usually check the console.

For in progress messages, you can just set the "in progress" message right before you start the web request.

Figuring out how to add something on error with a promise is left as an exercise to the reader. :)